cmd/loom/loom.go

```diff
@@ -1211,6 +1222,32 @@ func initQueryService(
 		return err
 	}
 
+	vmManager := vm.NewManager()
+	vmManager.Register(
+		vm.VMType_PLUGIN,
+		func(state loomchain.State) (vm.VM, error) {
+			v2ReceiptsEnabled := state.FeatureEnabled(loomchain.EvmTxReceiptsVersion2Feature, false)
+			receiptReader, err := receiptHandlerProvider.ReaderAt(state.Block().Height, v2ReceiptsEnabled)
+			if err != nil {
+				return nil, err
+			}
+			receiptWriter, err := receiptHandlerProvider.WriterAt(state.Block().Height, v2ReceiptsEnabled)
+			if err != nil {
+				return nil, err
+			}
+			return plugin.NewPluginVM(
+				loader,
+				state,
+				createRegistry(state),
+				&loomchain.DefaultEventHandler{},
+				log.Default,
+				newABMFactory,
+				receiptWriter,
+				receiptReader,
+			), nil
+		},
+	)
```

Comment:
Much of this is not needed to query EthCoin, see `createAddressMapperCtx` in `QueryServer` for an example of a minimal Go contract context. I don't think you need to change anything in `loom.go`.